src/main/java/org/apache/datasketches/tuple/AnotB.java

```diff
@@ -143,19 +142,58 @@ public void setA(final Sketch<S> skA) {
    *
    * @param skB The incoming Tuple sketch for the second (or following) argument <i>B</i>.
    */
+  @SuppressWarnings("unchecked")
   public void notB(final Sketch<S> skB) {
-    if (empty_ || skB == null || skB.isEmpty() || hashArr_ == null) { return; }
-    //skB is not empty
-    final long thetaLongB = skB.getThetaLong();
-    thetaLong_ = Math.min(thetaLong_, thetaLongB);
-
-    //process B
-    final DataArrays<S> daB = getResultArraysTuple(thetaLong_, curCount_, hashArr_, summaryArr_, skB);
-    hashArr_ = daB.hashArr;
-    summaryArr_ = daB.summaryArr;
+    if (skB == null) { return; } //ignore
 
-    curCount_ = hashArr_.length;
-    empty_ = curCount_ == 0 && thetaLong_ == Long.MAX_VALUE;
+    final long thetaLongB = skB.getThetaLong();
+    final int countB = skB.getRetainedEntries();
+    final boolean emptyB = skB.isEmpty();
+
+    final int id =
+        SetOperationCornerCases.createCornerCaseId(thetaLong_, curCount_, empty_, thetaLongB, countB, emptyB);
+    final CornerCase cCase = CornerCase.idToCornerCase(id);
+    final AnotbResult anotbResult = cCase.getAnotbResult();
+
+    switch (anotbResult) {
+      case NEW_1_0_T: {
+        reset();
+        break;
```

Comment:
Ah, an excellent observation and good question!  The New Degenerate {<1.0, 0, T} is a special temporary state in that it has never seen any data.  When the first item is presented to a New Degenerate sketch the T becomes F, and the count will only be changed to 1 if the hash of the item is <theta, which becomes {<1.0, 1, F}, otherwise it becomes {<1.0, 0, F}; both of which are stable states.

Because the New Degenerate sketch has never seen any data, it is interpreted as if it were just New {1.0, 0, T} in any set operation.

Reply:
Thanks for the additional details, I understand the reason now.